src/test_util/mod.rs

```diff
@@ -30,6 +32,68 @@ macro_rules! assert_downcast_matches {
     }};
 }
 
+#[macro_export]
+macro_rules! assert_within {
```

Comment:
We actually already have a similar crate pulled in for this: https://github.com/timberio/vector/blob/122778d90ffb7644e6811543696c845a1fb0417d/src/transforms/sampler.rs#L109

Reply:
While I could use that macro instead of this, the semantics are different (within is `Low <= Value <= High`, relative is `abs(Ideal - Actual) <= Epsilon`) I can rewrite it though if you think it's worth it.

Reply:
Nope, not a big deal. Just wanted to point it out in case you hadn't seen.